lib/gears/shape.lua

```diff
@@ -168,6 +220,243 @@ function module.isosceles_triangle(cr, width, height)
     cr:close_path()
 end
 
+--- A cross (**+**) symbol.
+--
+-- ![Shape example](@SHAPE_cross_SVG@)
+--
+-- @SHAPE_cross_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+-- @tparam[opt=width/3] number thickness The cross section thickness
+function module.cross(cr, width, height, thickness)
+    local thickness = thickness or width/3
+    local xpadding   = (width  - thickness) / 2
+    local ypadding   = (height - thickness) / 2
+    cr:move_to(xpadding, 0)
+    cr:line_to(width - xpadding, 0)
+    cr:line_to(width - xpadding, ypadding)
+    cr:line_to(width           , ypadding)
+    cr:line_to(width           , height-ypadding)
+    cr:line_to(width - xpadding, height-ypadding)
+    cr:line_to(width - xpadding, height         )
+    cr:line_to(xpadding        , height         )
+    cr:line_to(xpadding        , height-ypadding)
+    cr:line_to(0               , height-ypadding)
+    cr:line_to(0               , ypadding       )
+    cr:line_to(xpadding        , ypadding       )
+    cr:close_path()
+end
+
+--- A similar shape to the `rounded_rect`, but with sharp corners.
+--
+-- ![Shape example](@SHAPE_octogon_SVG@)
+--
+-- @SHAPE_octogon_EXAMPLE@
+--
+-- @param cr A cairo context
+-- @tparam number width The shape with
+-- @tparam number height The shape height
+-- @tparam number corner_radius 
+function module.octogon(cr, width, height, corner_radius)
+    local corner_radius = corner_radius or math.min(10, math.min(width, height)/4)
+    local offset = math.sqrt( (corner_radius*corner_radius) / 2 )
```

Comment:
Isn't this just `offset = corner_radius / math.sqrt(2)` (assuming `corner_radius` is positive)?
That doesn't really help me understand this (why not just `corner_radius` directly?), but at least feels easier to read.

Oh and: Why `math.min(10)` in the line before this? To me it looks like just `math.min(width,height/4)` should do.